tests/posterior_nn_test.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+from torch import zeros, ones, eye
+from torch.distributions import MultivariateNormal
+
+from sbi.inference.snpe.snpe_c import SnpeC
+from sbi.simulators.linear_gaussian import linear_gaussian
+
+
+def test_log_prob_with_different_x():
+
+    num_dim = 2
+
+    prior = MultivariateNormal(loc=zeros(num_dim), covariance_matrix=eye(num_dim))
+    posterior = SnpeC(simulator=linear_gaussian, prior=prior, x_o=ones(num_dim))(
+        num_rounds=1, num_simulations_per_round=1000
+    )
+
+    theta = posterior.sample(10)
+    posterior.log_prob(theta, x=None)
+    posterior.log_prob(theta, x=ones(num_dim))
+
+    with pytest.raises(ValueError):
+        # Reason: Multiple x passed.
```

Comment:
I would use the terminology batch_size > 1 whenever you talk of multiple x because it is more precise, any container (e.g. a list) would enable to pass multiple x but we check specifically for batch size.